src/metabase/util.clj

```diff
@@ -325,6 +325,13 @@
                  (and (re-matches #"^https?$" (.getProtocol url))           ; these are both automatically downcased
                       (re-matches #"^.+\..{2,}$" (.getAuthority url)))))))) ; this is the part like 'google.com'. Make sure it contains at least one period and 2+ letter TLD
 
+(defn nil-or-maplist?
+  "Is VALUE either nil or sequential? such that every item is a map?"
+  [v]
+  (or (nil? v)
+      (and (sequential? v)
+           (every? map? v))))
+
```

Comment:
The name of this function is a bit misleading, since vectors and arrays are both `sequential?` but neither one is a `List`.

I would suggest breaking this up into two functions, a more general `maybe?` function that checks for `nil` or satisfaction of a predicate:


```clojure
(defn maybe? [pred x]
  (or (nil? x)
      (pred x)))

(maybe? string? nil)   ; -> true
(maybe? string? "ABC") ; -> true
(maybve? string? 1)    ; -> false
```

And a more general function for checking whether something is a *sequence* of maps:

```clojure
(defn sequence-of-maps? [x]
  (and (sequential? x)
       (every? map? x)))

(sequence-of-maps? nil)               ; -> false
(sequence-of-maps? [{:a :b} {:c :d}]) ; -> true
```

`sequence-of-maps?` is a more accurate name. The two can then be used generally or combined to get the desired effect:

```clojure
(maybe? sequence-of-maps? nil)        ; -> true
(maybe? sequence-of-maps? '({:a :b})) ; -> true
(maybe? sequence-of-maps? [1 2 3])    ; -> false
```